web/src/components/tasks/cigarette-license/CigaretteLicense.tsx

```diff
@@ -94,6 +101,44 @@ export const CigaretteLicense = (props: Props): ReactElement => {
     },
   ];
 
+  const handlePaymentSuccess = useCallback(() => {
+    if (!updateQueue || !business) return;
+
+    queueUpdateTaskProgress("cigarette-license", "COMPLETED");
+    updateQueue
+      ?.queueBusiness({
+        ...updateQueue.currentBusiness(),
+        cigaretteLicenseData: {
+          ...business.cigaretteLicenseData,
+          lastUpdatedISO: new Date(Date.now()).toISOString(),
+          paymentInfo: {
+            ...business.cigaretteLicenseData?.paymentInfo,
+            paymentComplete: true,
+          },
+        },
+      })
+      .update();
+  }, [updateQueue, business, queueUpdateTaskProgress]);
+
+  useEffect(() => {
+    if (!router?.isReady) return;
+
+    if (
+      checkQueryValue(router, QUERIES.completePayment, "failure") ||
+      checkQueryValue(router, QUERIES.completePayment, "duplicate")
+    ) {
+      setStepIndex(3);
+      setSubmissionError("PAYMENT");
+    }
+    if (checkQueryValue(router, QUERIES.completePayment, "cancel")) {
+      setStepIndex(3);
+    }
+    if (checkQueryValue(router, QUERIES.completePayment, "success")) {
+      handlePaymentSuccess();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
We discussed this line, I agree this is an acceptable exception for the eslint disabling here